hazelcast/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -290,6 +290,9 @@ public void startScanningForJobs() {
                 jobRepository.putNewJobRecord(jobRecord);
                 logger.info("Starting job " + idToString(masterContext.jobId()) + " based on submit request");
             } catch (Throwable e) {
+                jetServiceBackend.getJobClassLoaderService()
+                                 .tryRemoveClassloadersForJob(jobId, COORDINATOR);
+
                 res.completeExceptionally(e);
                 throw e;
             } finally {
```

Comment:
Why not in the `finally`?

Reply:
Good question. The reason is that we want to do this only for the exceptional case where we don't go into `tryStartJob`.